### DISCUSSION TITLE
Web Dev for Beginners inside Canvas LMS!

### DISCUSSION THREAD

**Discussion by kyletower** [microsoft/Web-Dev-For-Beginners#818] [Teacher corner]
**Hello, fellow educators! 👋🏼**

I have been working on bringing all of this content into the **Canvas LMS** so that I, and any other instructor who uses Canvas, can benefit from this excellent curriculum.

I've spent about 127 hours so far and accomplished the following 😅

✅ All lessons/lectures/content pages are hosted inside a Canvas page which is ordered inside of a Module.
✅ All pre/post lecture quizzes are hosted inside Canvas and are auto-graded providing students (and teachers) with instant feedback. These are also in order inside of a Module.
✅ All assignments are hosted inside a Canvas assignment. Again, these are ordered inside a Module.
✅ Many of the pages have embedded interactive code snippets so the learner doesn't have to leave the Canvas environment.
✅ I've added some "Meet the Professional" pages to the modules derived from KhanAcademy.org.
✅ I've added multiple Sample Solutions for assignments into the course.

![web dev - modules - 1 preview](https://user-images.githubusercontent.com/9216820/194667887-96801386-90dc-43c3-8143-8ad591fc5ade.png)

![web dev - modules - all](https://user-images.githubusercontent.com/9216820/194667893-5bc123fe-e5f2-461c-b1e1-c092a9905723.png)

If you have any further ideas of how I can improve the Canvas LMS version of this course, let me know. 

I'll be submitting a Pull Request for an updated version of the *.imscc file so you can import it into your own Canvas instance and use it with your students.

--
grace & peace,
[ kyle tower ]

**Comment by Jesuscolum** [comment]
> **Hello, fellow educators! 👋🏼**
> 
> I have been working on bringing all of this content into the **Canvas LMS** so that I, and any other instructor who uses Canvas, can benefit from this excellent curriculum.
> 
> I've spent about 127 hours so far and accomplished the following 😅
> 
> ✅ All lessons/lectures/content pages are hosted inside a Canvas page which is ordered inside of a Module. ✅ All pre/post lecture quizzes are hosted inside Canvas and are auto-graded providing students (and teachers) with instant feedback. These are also in order inside of a Module. ✅ All assignments are hosted inside a Canvas assignment. Again, these are ordered inside a Module. ✅ Many of the pages have embedded interactive code snippets so the learner doesn't have to leave the Canvas environment. ✅ I've added some "Meet the Professional" pages to the modules derived from KhanAcademy.org. ✅ I've added multiple Sample Solutions for assignments into the course.
> 
> ![web dev - modules - 1 preview](https://user-images.githubusercontent.com/9216820/194667887-96801386-90dc-43c3-8143-8ad591fc5ade.png)
> 
> ![web dev - modules - all](https://user-images.githubusercontent.com/9216820/194667893-5bc123fe-e5f2-461c-b1e1-c092a9905723.png)
> 
> If you have any further ideas of how I can improve the Canvas LMS version of this course, let me know.
> 
> I'll be submitting a Pull Request for an updated version of the *.imscc file so you can import it into your own Canvas instance and use it with your students.
> 
> -- grace & peace, [ kyle tower ]

**Reply by its0Vivek** [reply]
How can I get it?

**Reply by kyletower** [reply]
I have just uploaded it to [my forked repo here](https://github.com/kyletower/Web-Dev-For-Beginners/blob/main/teaching-files/canvas-course-web-dev-for-beginners-adapted-by-kyle-tower-compressed.imscc).

**Comment by bluesky8885** [comment]
Can I add it as a lesson in Classroom?

**Reply by kyletower** [reply]
Unfortunately, I don't believe the Canvas IMSCC export is compatible with Google Classroom.
